app/BuildMetric.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+namespace App;
+
+use Backpack\CRUD\app\Models\Traits\CrudTrait;
+use Illuminate\Database\Eloquent\Model;
+
+class BuildMetric extends Model
+{
+    use CrudTrait;
+    protected $guarded = ['id'];
+
+    public function gun()
+    {
+        return $this->belongsTo(Gun::class, 'gun_id');
+    }
+
+    public function character()
+    {
+        return $this->belongsTo(Character::class, 'character_id');
+    }
+
+    public function patch()
+    {
+        return $this->belongsTo(Patch::class);
+    }
+
+    public function getModMatrix($gun, $combo)
+    {
+        $selected_index = [];
+        $gun_object = $gun;
+        $gun_mods = $gun_object->mods->groupBy('mod_tier');
+
+        $combo_array = str_split($combo);
+
+        foreach ($combo_array as $key => $tier) {
+            $selected = false;
+            if ($tier != '-') {
+                $selected = true;
+            }
+            // dd($tier);
+            $selected_index[$key + 1] = ['value' => $tier, 'selected' => $selected];
+        }
+        $matrixArray = compact('gun_mods', 'selected_index');
+
+        return $matrixArray;
+    }
+
+    public function getOverclock($gun, $index)
+    {
+        $gun_object = Gun::findOrFail($gun);
+        $overclock = Overclock::where([
+            ['character_id', '=', $gun_object->character_id],
```

Comment:
Pretty sure we can remove this as `gun_id` is a 1 to 1 with `overclock` in this case.

Reply:
Please explain

Reply:
In this case, `character` is superfluous.

Scout is the only class that has a GK2, the GK2 is the only gun that has Electrifying Reload. No need to specify that the overclock Electrifying Reload is a Scout overclock.